src/main/java/banana/Duke.java

```diff
@@ -0,0 +1,352 @@
+        }
+    }
+
+    /**
+     * Runs the program.
+     */
+    public void run() {
+        try {
+            Parser.displayLabel(WELCOME_LABEL);
+            String input = ui.getInput();
+            while (!input.equals("bye")) {
+                if (!input.equals("")) {
+                    Parser p = new Parser(input);
+                    p.useInput(tasks);
+                    writeToFile(input, tasks);
+                }
+                input = ui.getInput();
+            }
+            Parser.displayLabel(BYE_LABEL);
+        } catch (DukeException | IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void main(String[] args) {
+        /*String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);*/
+        new Duke("/Users/ravi57004/ip/src/main/java/Tasks.txt").run();
+    }
+
+    /**
+     * Writes to the Tasks.txt file everytime
+     * a change in tasks has occurred.
+     *
+     * @param input user input.
+     * @param tasks the tasks to be managed.
+     * @throws IOException if not able to write to the file.
+     */
+    public static void writeToFile(String input, TaskList tasks) throws IOException {
+        String text = "";
+        FileWriter fw = new FileWriter(
+                "/Users/ravi57004/ip/src/main/java/Tasks.txt", false);
+        if (!input.equals("bye")) {
+            for (int i = 0; i < tasks.size(); i++) {
+                String doneStr = "No";
+                if (tasks.getTask(i).getIsDone().equals("[X]")) {
+                    doneStr = "Yes";
+                }
+                if (tasks.getTask(i) instanceof ToDo) {
+                    text += "T ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + "\n";
+                } else if (tasks.getTask(i) instanceof Deadline) {
+                    Deadline dl = (Deadline) tasks.getTask(i);
+                    text += "D ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + " ~ " + dl.getDeadLine() + "\n";
+                } else if (tasks.getTask(i) instanceof Event) {
+                    Event ev = (Event) tasks.getTask(i);
+                    text += "E ~ " + doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + " ~ " + ev.getEvent() + "\n";
+                } else {
+                    text += doneStr + " ~ " +
+                            tasks.getTask(i).getDescription() + "\n";
+                }
+            }
+        }
+        fw.write(text);
+        fw.close();
+    }
+
+}
+
+/**
+ * The DukeException class
+ * throws specialised Duke exceptions.
+ *
+ * @author: Ravi Ananya
+ **/
+class DukeException extends Exception {
+
+    private String text;
```

Comment:
Perhaps it would be better if a more descriptive name (e.g. errorMessage) was used for this variable?